annMap无法正确解析Kotlin中的组合注解

`it.annMap("io.swagger.annotations.ApiImplicitParams")`在解析`kotlin`中的组合注解时，内层注解无法正确解析。
错误信息: `javax.script.ScriptException: javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: allowableValues for class: java.lang.String`

相关代码：
```kotlin
    @ApiOperation("创建商圈")
    @ApiImplicitParams(value = [
        ApiImplicitParam(name = "name", value = "商圈名称", required = true)]
    )
    @ApiResponses(value = [ApiResponse(code = 200, message = "创建的id")])
    @PostMapping
    fun create(@RequestBody @Validated(InsertGroup::class) @ApiParam(hidden = true) dto: ShopAreaDto): Long {
        return shopAreaAppService.createArea(dto.toBo(), this.currentUser()!!.userId)
    }
```
相关脚本：
```
def maps = it.annMap("io.swagger.annotations.ApiImplicitParams")
for(map in maps.value){
    def desc = map.value
    def allowableValues = map.allowableValues
    if(tool.notNullOrBlank(allowableValues)){
        desc = desc + "n允许的值:" + allowableValues
    }
    def paramType = map.paramType ?: "query"
    def required = map.required?:true
    required = required == "true"
    if(paramType==""||paramType=="query"){
        api.setParam(map.name,map.defaultValue,required,desc)
    }else if(paramType=="form"){
        api.setFormParam(map.name,map.defaultValue,required,desc)
    }else if(paramType=="path"){
        api.setPathParam(map.name,map.defaultValue,desc)
    }else if(paramType=="header"){
        api.setHeader(map.name,map.defaultValue,required,desc)
    }
}
```
完整堆栈信息:
```text
[TRACE]	javax.script.ScriptException: javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: allowableValues for class: java.lang.String
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:158)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.eval(ScriptRuleParser.kt:89)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.access$eval(ScriptRuleParser.kt:34)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseStringRule$1.invoke(ScriptRuleParser.kt:68)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseStringRule$1.invoke(ScriptRuleParser.kt:34)
	at com.itangcent.intellij.config.rule.StringRule$Companion$of$1.compute(StringRule.kt:8)
	at com.itangcent.intellij.config.rule.StringRule$Companion$of$1.compute(StringRule.kt:6)
	at com.itangcent.intellij.config.rule.StringRule$Companion$filterWith$1.invoke(StringRule.kt:16)
	at com.itangcent.intellij.config.rule.StringRule$Companion$filterWith$1.invoke(StringRule.kt:4)
	at com.itangcent.intellij.config.rule.StringRule$Companion$of$1.compute(StringRule.kt:8)
	at com.itangcent.intellij.config.rule.StringRule$Companion$of$1.compute(StringRule.kt:6)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$compute$1.apply(DefaultRuleComputer.kt:64)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$compute$1.apply(DefaultRuleComputer.kt:9)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1631)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.compute(DefaultRuleComputer.kt:66)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.computer(DefaultRuleComputer.kt:30)
	at com.itangcent.intellij.config.rule.RuleComputerKt.computer(RuleComputer.kt:35)
	at com.itangcent.idea.plugin.api.export.SpringRequestClassExporter.processCompleted(SpringRequestClassExporter.kt:262)
	at com.itangcent.idea.plugin.api.export.yapi.YapiSpringRequestClassExporter.processCompleted(YapiSpringRequestClassExporter.kt:26)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.exportMethodApi(AbstractRequestClassExporter.kt:174)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.access$exportMethodApi(AbstractRequestClassExporter.kt:36)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter$export$1.invoke(AbstractRequestClassExporter.kt:127)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter$export$1.invoke(AbstractRequestClassExporter.kt:36)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter$sam$java_util_function_Consumer$0.accept(AbstractRequestClassExporter.kt)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.foreachMethod(AbstractRequestClassExporter.kt:453)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.export(AbstractRequestClassExporter.kt:123)
	at com.itangcent.idea.plugin.api.export.ComboClassExporter$export$1.apply(ComboClassExporter.kt:41)
	at com.itangcent.idea.plugin.api.export.ComboClassExporter$export$1.apply(ComboClassExporter.kt:15)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.itangcent.idea.plugin.api.export.ComboClassExporter.export(ComboClassExporter.kt:41)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$ApiExporterAdapter.doExportApisFromMethod(SuvApiExporter.kt:317)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$ApiExporterAdapter.access$doExportApisFromMethod(SuvApiExporter.kt:164)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$ApiExporterAdapter$exportApisFromMethod$1$1$1$1.invoke(SuvApiExporter.kt:222)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$ApiExporterAdapter$exportApisFromMethod$1$1$1$1.invoke(SuvApiExporter.kt:164)
	at com.itangcent.intellij.context.ActionContext$runInReadUI$1.run(ActionContext.kt:324)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:52)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:869)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:51)
	at com.itangcent.intellij.context.ActionContext.runInReadUI(ActionContext.kt:318)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$ApiExporterAdapter$exportApisFromMethod$1$1$1.invoke(SuvApiExporter.kt:220)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$ApiExporterAdapter$exportApisFromMethod$1$1$1.invoke(SuvApiExporter.kt:164)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$YapiApiExporterAdapter.beforeExport(SuvApiExporter.kt:487)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$ApiExporterAdapter$exportApisFromMethod$1$1.invoke(SuvApiExporter.kt:219)
	at com.itangcent.idea.plugin.api.export.suv.SuvApiExporter$ApiExporterAdapter$exportApisFromMethod$1$1.invoke(SuvApiExporter.kt:164)
	at com.itangcent.intellij.context.ActionContext$runAsync$2.run(ActionContext.kt:173)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: allowableValues for class: java.lang.String
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:320)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:155)
	... 79 more
Caused by: groovy.lang.MissingPropertyException: No such property: allowableValues for class: java.lang.String
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:65)
	at org.codehaus.groovy.runtime.callsite.GetEffectivePojoPropertySite.getProperty(GetEffectivePojoPropertySite.java:65)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGetProperty(AbstractCallSite.java:298)
	at Script216.run(Script216.groovy:5)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:317)
	... 80 more
```

--- a/common-api/build.gradle
+++ b/common-api/build.gradle
@@ -28,7 +28,7 @@ dependencies {
 
     compile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
 
-    compile('com.itangcent:commons:0.7.7-SNAPSHOT') {
+    compile('com.itangcent:commons:0.7.8-SNAPSHOT') {
         exclude group: 'com.google.inject'
         exclude group: 'com.google.code.gson'
     }

--- a/idea-plugin/build.gradle
+++ b/idea-plugin/build.gradle
@@ -33,10 +33,9 @@ intellij {
     updateSinceUntilBuild false
     sameSinceUntilBuild false
 
-    plugins 'org.jetbrains.kotlin:1.3.11-release-IJ2017.3-1'
+    plugins "org.jetbrains.kotlin:1.3.11-release-IJ2017.3-1", "org.intellij.scala:2017.3.11"
 //    plugins 'org.jetbrains.kotlin:1.3.21-release-IJ2018.2-1' //dependency kotlin
 
-    plugins 'org.intellij.scala:2017.3.11'
 
 //    plugins = ['java']
 }
@@ -100,17 +99,17 @@ dependencies {
 //    compile fileTree(dir: 'libs', include: ['*.jar'])
 
 
-    compile('com.itangcent:intellij-idea:0.7.7-SNAPSHOT') {
+    compile('com.itangcent:intellij-idea:0.7.8-SNAPSHOT') {
         exclude group: 'com.google.inject'
         exclude group: 'com.google.code.gson'
     }
 
-    compile('com.itangcent:intellij-kotlin-support:0.7.7-SNAPSHOT') {
+    compile('com.itangcent:intellij-kotlin-support:0.7.8-SNAPSHOT') {
         exclude group: 'com.google.inject'
         exclude group: 'com.google.code.gson'
     }
 
-    compile('com.itangcent:intellij-scala-support:0.7.7-SNAPSHOT') {
+    compile('com.itangcent:intellij-scala-support:0.7.8-SNAPSHOT') {
         exclude group: 'com.google.inject'
         exclude group: 'com.google.code.gson'
     }

--- a/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/GroovyRuleParser.kt
+++ b/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/GroovyRuleParser.kt
@@ -8,5 +8,6 @@ class GroovyRuleParser : StandardJdkRuleParser() {
     override fun scriptType(): String {
         return "groovy"
     }
+
 }
 

--- a/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/ScriptRuleParser.kt
+++ b/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/ScriptRuleParser.kt
@@ -69,6 +69,12 @@ abstract class ScriptRuleParser : RuleParser {
         }
     }
 
+    override fun parseEventRule(rule: String): EventRule? {
+        return EventRule.of {
+            eval(rule, it)
+        }
+    }
+
     private fun eval(ruleScript: String, context: RuleContext): Any? {
         return try {
             val simpleScriptContext = SimpleScriptContext()

--- a/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/SuvRuleParser.kt
+++ b/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/SuvRuleParser.kt
@@ -40,6 +40,15 @@ class SuvRuleParser : RuleParser {
         }
     }
 
+    override fun parseEventRule(rule: String): EventRule? {
+        return when {
+            rule.isBlank() -> null
+            rule.startsWith(JS_PREFIX) -> getRuleParser(JsRuleParser::class).parseEventRule(rule.removePrefix(JS_PREFIX))
+            rule.startsWith(GROOVY_PREFIX) -> getRuleParser(GroovyRuleParser::class).parseEventRule(rule.removePrefix(GROOVY_PREFIX))
+            else -> getRuleParser(SimpleRuleParser::class).parseEventRule(rule)
+        }
+    }
+
     private val ruleParserCache: HashMap<KClass<*>, RuleParser> = LinkedHashMap()
 
     private fun getRuleParser(parserClass: KClass<*>): RuleParser {

